pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -127,3 +127,22 @@ def clip_grad_by_norm(self, optimizer: Optimizer, clip_val: Union[int, float]) -
         """Clip gradients by norm"""
         parameters = self.master_params(optimizer)
         torch.nn.utils.clip_grad_norm_(parameters, clip_val)
+
+    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        """
+        Called when Trainer is restoring from a checkpoint.
+
+        Args:
+            checkpoint: The full checkpoint content being restored.
+        """
+        pass
+
+    def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        """
+        Called when Trainer is saving a checkpoint.
+
+        Args:
+            checkpoint: The full checkpoint dictionary before it gets dumped to a file.
+                Implementations of this hook can insert additional data into this dictionary.
+        """
+        pass
```

Comment:
why not use the CheckpointHooks interface? https://github.com/PyTorchLightning/pytorch-lightning/blob/ca89a7f3443710a2313e97b91ac908867eb33be5/pytorch_lightning/core/hooks.py#L782

Reply:
sure, I could add this in as a parent class if that's the intended use case